TLM/TLM/LoadingExtension.cs

```diff
@@ -162,93 +168,82 @@ public class LoadingExtension : LoadingExtensionBase {
-                                    "a newer game version ({0}) than TM:PE has been built for ({1}). " +
-                                    "Please be aware that TM:PE has not been updated for the newest game " +
-                                    "version yet and thus it is very likely it will not work as expected.",
-                                    BuildConfig.applicationVersion,
-                                    BuildConfig.VersionToString(TrafficManagerMod.GAME_VERSION, false));
-
-                                Log.Error(msg);
-                                Singleton<SimulationManager>.instance.m_ThreadingWrapper.QueueMainThread(
-                                        () => {
-                                            UIView.library
-                                                  .ShowModal<ExceptionPanel>("ExceptionPanel")
-                                                  .SetMessage(
-                                                      "TM:PE has not been updated yet",
-                                                      msg,
-                                                      false);
-                                        });
-                            } else if (isModNewer) {
-                                string msg = string.Format(
-                                    "Traffic Manager: President Edition has been built for game version {0}. " +
-                                    "You are running game version {1}. Some features of TM:PE will not " +
-                                    "work with older game versions. Please let Steam update your game.",
-                                    BuildConfig.VersionToString(TrafficManagerMod.GAME_VERSION, false),
-                                    BuildConfig.applicationVersion);
-
-                                Log.Error(msg);
-                                Singleton<SimulationManager>
-                                    .instance.m_ThreadingWrapper.QueueMainThread(
-                                        () => {
-                                            UIView.library
-                                                  .ShowModal<ExceptionPanel>("ExceptionPanel")
-                                                  .SetMessage(
-                                                      "Your game should be updated",
-                                                      msg,
-                                                      false);
-                                        });
-                            }
-                        }
-
-                        IsGameLoaded = true;
-                        break;
-                    }
-
-                default: {
-                        Log.Info($"OnLevelLoaded: Unsupported game mode {mode}");
-                        return;
-                    }
+            if (BuildConfig.applicationVersion != BuildConfig.VersionToString(
```

Comment:
I really appreciate when big code chunks are moved out to functions with good names.
This one could be named something like `OnGameVersionMismatch` or similar?

Reply:
this is not a code I have changed and if you hide whitespaces you wowuld notice that.

Reply:
but yeah I agree with you this place does need some cleanup :)